ui/core/src/hooks/addLiquidityCalculator.ts

```diff
@@ -233,6 +236,22 @@ export function usePoolCalculator(input: {
     return bPerARatioProjected.value.toFixed(8);
   });
 
+  effect(() => {
+    // if in guided mode
+    // calculate the price ratio of A / B
+    if (input.asyncPooling.value && input.lastFocusedTokenField.value !== null) {
+      if (bPerARatio === null || aPerBRatio === null) {
+        return null;
+      }
+      if (input.lastFocusedTokenField.value === 'A') {
+        input.tokenBAmount.value = (Number(input.tokenAAmount.value) * Number(bPerARatio.value.toFixed(8))).toString();
```

Comment:
Probably don't want to use `Number` we would be better off using `AssetAmount` instead.